src/containers/Application/ApplicationPage.tsx

```diff
@@ -1,34 +1,100 @@
 import React, { useEffect, useState } from 'react'
-import { FullStructure, ResponsesByCode, ElementStateNEW } from '../../utils/types'
+import {
+  FullStructure,
+  ResponsesByCode,
+  ElementStateNEW,
+  MethodToCallOnRevalidation,
+  SectionAndPage,
+} from '../../utils/types'
 import useGetFullApplicationStructure from '../../utils/hooks/useGetFullApplicationStructure'
 import { ApplicationStatus } from '../../utils/generated/graphql'
-import { useApplicationState } from '../../contexts/ApplicationState'
 import { useUserState } from '../../contexts/UserState'
 import { useRouter } from '../../utils/hooks/useRouter'
-import { Loading, NoMatch } from '../../components'
+import { Loading } from '../../components'
 import strings from '../../utils/constants'
-import messages from '../../utils/messages'
 import { Button, Grid, Header, Message, Segment, Sticky } from 'semantic-ui-react'
 import { PageElements } from '../../components/Application'
+import { useFormElementUpdateTracker } from '../../contexts/FormElementUpdateTrackerState'
 
 interface ApplicationProps {
   structure: FullStructure
   responses?: ResponsesByCode
 }
 
+const getFirstInvalidPage: (fullStructure: FullStructure) => SectionAndPage | null = (
+  fullStructure
+) => {
```

Comment:
I think that using the typing here is confusing, maybe best to leave as:
```suggestion
const getFirstInvalidPage = ( fullStructure: FullStructure ):  SectionAndPage | null => {
```

Reply:
much better thanks